core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -486,18 +487,24 @@ private StartupConsistencyChecker(Long fileId) {
             inconsistentUris.add(parentUri);
           }
           for (Inode childInode : ((InodeDirectory) parentInode).getChildren()) {
-            AlluxioURI childUri = parentUri.join(childInode.getName());
-            if (childInode.isDirectory()) {
-              dirsToCheck.add(childInode.getId());
-            } else {
-              childInode.lockRead();
-              try {
+            try {
+              childInode.lockReadAndCheckParent(parentInode);
```

Comment:
One issue I noticed with our current master code is that we do not always abstract the inode lock. Ideally we should always use `LockedInodePath` (which is an `AutoClosable`), but in certain places we directly use the low level API (ie. `inode.lockRead()`) which gets us into these less elegant pieces of code. I suggest we refactor these instances in a separate PR instead of tagging it onto this one, since I think we would need to expand the utility provided in LockedInodePath.